### ISSUE TITLE
Detection of redhat problematic  (perhaps more accurate to say "redhat problematic")

### ISSUE THREAD

**Issue by hanleybrand** [jordansissel/pleaserun#72]
tldr:  The only reliable way I could get pleaserun to work with RHEL 6.6 was to install ruby 2.1.5 via `rvm` and use `rvmsudo` to run the command from the gem's bin directory : 

```
cd /usr/local/rvm/gems/ruby-2.1.5/gems/pleaserun-0.0.9/bin    
rvmsudo ./pleaserun  -p sysv -v lsb-3.1 --install /opt/kibana/bin/kibana
```

which, once I got there worked perfectly ;-)

-----

More info:

Redhat 6.6 ships with no ruby, and yum will install Ruby 1.8, so to use pleaserun you need to otherwise install. I don't know ruby myself (I found this repo via elasticsearch/kibana#1595) and saw the note in another issue regarding postmodern/ruby-install so maybe that would have been easier)

Regardless, it seems like RHEL is harder to get working than say Ubuntu. Anyway, below is the error from running without specifying sysv/lsb-3.1:

```
$ rvmsudo ./pleaserun --install /opt/kibana/bin/kibana
/usr/local/rvm/gems/ruby-2.1.5/gems/pleaserun-0.0.9/lib/pleaserun/detector.rb:46:in `detect': redhat 6.6 (PleaseRun::Detector::UnknownSystem)
```

I saw jordansissel/pleaserun#60 but I had already installed 0.9.0 and the last step `cp pleaserun-0.0.9/templates/sysv/lsb-3.1/* pleaserun-0.0.9/templates/sysv/` didn't fix the issue in my case, mainly because as far as I can see in the repo 'redhat' isn't defined anywhere?

**Comment by jordansissel** [comment]
hmm, adding support for rhel should be as trivial as updating the distro-to-platform mapping: https://github.com/jordansissel/pleaserun/blob/master/lib/pleaserun/detector.rb#L10-L29

However, this doesn't have to be  a blocker for you. You don't need to rely on the detection. You can specify the platform to use. On RHEL6, the init system is Upstart 0.6.5, so you can use `pleaserun --install -p upstart -v 0.6.5 ...`

> Redhat 6.6 ships with no ruby, and yum will install Ruby 1.8,

Yeah :(

Red Hat and Debian both generally ship ancient versions of software (Ruby 1.8 has been dead for years, Ruby 1.9.3 was just declared dead 7 days ago). I'm willing to try and support them, but I have no resources with which to test this right now, so it's difficult to do the work myself. Despite the resourcing limitation, I am open to supporting older rubies just like I do with fpm.

**Comment by aecolley** [comment]
Here's what happened when I tried to install pleaserun on CentOS 6.8. If you are on the Docker bandwagon, it's a one-liner that might help you figure out support for old rubies on such systems. I'm going to investigate the easiest way to get a newer ruby onto centos6.

```
bash-4.3$ docker run --rm -it centos:6 /bin/sh -c "yum install -y rubygems && gem install pleaserun"
Loaded plugins: fastestmirror
Setting up Install Process
base                                                     | 3.7 kB     00:00
base/primary_db                                          | 4.7 MB     00:00
cr                                                       | 3.3 kB     00:00
cr/primary_db                                            | 1.2 kB     00:00
extras                                                   | 3.4 kB     00:00
extras/primary_db                                        |  36 kB     00:00
updates                                                  | 3.4 kB     00:00
updates/primary_db                                       | 729 kB     00:00
Resolving Dependencies
--> Running transaction check
---> Package rubygems.noarch 0:1.3.7-5.el6 will be installed
--> Processing Dependency: ruby(abi) = 1.8 for package: rubygems-1.3.7-5.el6.noarch
--> Processing Dependency: ruby-rdoc for package: rubygems-1.3.7-5.el6.noarch
--> Processing Dependency: /usr/bin/ruby for package: rubygems-1.3.7-5.el6.noarch
--> Running transaction check
---> Package ruby.x86_64 0:1.8.7.374-4.el6_6 will be installed
---> Package ruby-libs.x86_64 0:1.8.7.374-4.el6_6 will be installed
--> Processing Dependency: libreadline.so.5()(64bit) for package: ruby-libs-1.8.7.374-4.el6_6.x86_64
---> Package ruby-rdoc.x86_64 0:1.8.7.374-4.el6_6 will be installed
--> Processing Dependency: ruby-irb = 1.8.7.374-4.el6_6 for package: ruby-rdoc-1.8.7.374-4.el6_6.x86_64
--> Running transaction check
---> Package compat-readline5.x86_64 0:5.2-17.1.el6 will be installed
---> Package ruby-irb.x86_64 0:1.8.7.374-4.el6_6 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package                Arch         Version                   Repository  Size
================================================================================
Installing:
 rubygems               noarch       1.3.7-5.el6               base       207 k
Installing for dependencies:
 compat-readline5       x86_64       5.2-17.1.el6              base       130 k
 ruby                   x86_64       1.8.7.374-4.el6_6         base       538 k
 ruby-irb               x86_64       1.8.7.374-4.el6_6         base       317 k
 ruby-libs              x86_64       1.8.7.374-4.el6_6         base       1.7 M
 ruby-rdoc              x86_64       1.8.7.374-4.el6_6         base       381 k

Transaction Summary
================================================================================
Install       6 Package(s)

Total download size: 3.2 M
Installed size: 11 M
Downloading Packages:
(1/6): compat-readline5-5.2-17.1.el6.x86_64.rpm          | 130 kB     00:00
(2/6): ruby-1.8.7.374-4.el6_6.x86_64.rpm                 | 538 kB     00:00
(3/6): ruby-irb-1.8.7.374-4.el6_6.x86_64.rpm             | 317 kB     00:00
(4/6): ruby-libs-1.8.7.374-4.el6_6.x86_64.rpm            | 1.7 MB     00:00
(5/6): ruby-rdoc-1.8.7.374-4.el6_6.x86_64.rpm            | 381 kB     00:00
(6/6): rubygems-1.3.7-5.el6.noarch.rpm                   | 207 kB     00:00
--------------------------------------------------------------------------------
Total                                           1.3 MB/s | 3.2 MB     00:02
warning: rpmts_HdrFromFdno: Header V3 RSA/SHA1 Signature, key ID c105b9de: NOKEY
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
Importing GPG key 0xC105B9DE:
 Userid : CentOS-6 Key (CentOS 6 Official Signing Key) <centos-6-key@centos.org>
 Package: centos-release-6-7.el6.centos.12.3.x86_64 (@CentOS/6.7)
 From   : /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : compat-readline5-5.2-17.1.el6.x86_64                         1/6
  Installing : ruby-libs-1.8.7.374-4.el6_6.x86_64                           2/6
  Installing : ruby-1.8.7.374-4.el6_6.x86_64                                3/6
  Installing : ruby-irb-1.8.7.374-4.el6_6.x86_64                            4/6
  Installing : ruby-rdoc-1.8.7.374-4.el6_6.x86_64                           5/6
  Installing : rubygems-1.3.7-5.el6.noarch                                  6/6
  Verifying  : ruby-rdoc-1.8.7.374-4.el6_6.x86_64                           1/6
  Verifying  : compat-readline5-5.2-17.1.el6.x86_64                         2/6
  Verifying  : ruby-1.8.7.374-4.el6_6.x86_64                                3/6
  Verifying  : ruby-irb-1.8.7.374-4.el6_6.x86_64                            4/6
  Verifying  : rubygems-1.3.7-5.el6.noarch                                  5/6
  Verifying  : ruby-libs-1.8.7.374-4.el6_6.x86_64                           6/6

Installed:
  rubygems.noarch 0:1.3.7-5.el6

Dependency Installed:
  compat-readline5.x86_64 0:5.2-17.1.el6  ruby.x86_64 0:1.8.7.374-4.el6_6
  ruby-irb.x86_64 0:1.8.7.374-4.el6_6     ruby-libs.x86_64 0:1.8.7.374-4.el6_6
  ruby-rdoc.x86_64 0:1.8.7.374-4.el6_6

Complete!
ERROR:  Error installing pleaserun:
	mustache requires Ruby version >= 1.9.3.
bash-4.3$
```

**Comment by aecolley** [comment]
It turns out the only easy way is to install rvm as described on [rvm.io](http://rvm.io/) and then `rvm install 2.3`.

**Comment by Sudhan2612** [comment]
I have tried to execute the kibana as background service through initialization script for RHEL https://github.com/cjcotton/init-kibana. 

I was able to the message that service is started through /etc/init.d.

 But service is not actually starting in background.
